hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -72,6 +72,18 @@ public static String getNestedFieldValAsString(GenericRecord record, String fiel
     return (obj == null) ? null : obj.toString();
   }
 
+  /**
+   * Obtain value of the provided field, denoted by dot notation. e.g: a.b.c. Return null if not found
+   */
+  public static Object getNestedFieldValOrNull(GenericRecord record, String fieldName) {
+    try {
+      return getNestedFieldVal(record, fieldName);
+    } catch (HoodieException e) {
```

Comment:
Not sure if handling this via Exception handling is the most efficient way?  `_hoodie_delete_marker` is just a top level field right? So, can't we call `record.get()` and check for null. I think it will be much less overhead?